src/gasman.c

```diff
@@ -2081,54 +2081,65 @@ void CheckMasterPointers( void )
 }
 
 
-/****************************************************************************
-**
-*F  SwapMasterPoint( <bag1>, <bag2> ) . . . swap pointer of <bag1> and <bag2>
-*/
-void SwapMasterPoint (
-    Bag                 bag1,
-    Bag                 bag2 )
+// Swap the master pointers of bag1 and bag2
+//
+// We need to make sure the correct bags are GCed. There are various things to
+// consider.
+//
+// As changed bags are stored as a singly linked list, we can't easily take
+// anything out of the list of changed bags. Also, as the changed bags are stored
+// in a linked list with the pointer stored inside the bag itself, when we swap
+// PTR_BAG, we swap LINK_BAG as well.
+//
+// If both bags are old, if the original bag1 was changed we need to now mark
+// bag2 changed (and vice-versa).
+//
+// If both bags are young, GAP assumes they are both changed.
```

Comment:
I think this is quite confusing to read... what does it mean that "both bags are changed" ? I think that's not the relevant thing we need to convey here. Indeed, it is not quite clear to me *what* the message here is. Perhaps: "If both bags are young, there is no need to either of them into the list of changed bags" ? But that would be kind of redundant or at least not helpful information (at least at this point), so I'd omit it.

But perhaps you meant something else?